pkg/codegen/testing/test/program_driver.go

```diff
@@ -249,6 +250,9 @@ func TestProgramCodegen(
 			}
 
 			expectedFile := filepath.Join(testDir, tt.Directory+"."+testcase.Extension)
+			if tt.DependencyFile != "" {
```

Comment:
Mutually exclusive fields like this scare me. It would be ideal if we separated the version test framework from a `ProgramTest`. If we don't, can we add an error for specifying both `testcase.Extension` and `testcase.DependencyFile`.